libs/wallet/src/web3-react/updater.ts

```diff
@@ -86,19 +90,39 @@ function _useSafeInfo(walletInfo: WalletInfo): GnosisSafeInfo | undefined {
       }
     }
 
-    const interval = setInterval(update, SAFE_INFO_UPDATE_INTERVAL)
+    const updateSafeAppsSdkInfo = async () => {
+      if (safeAppsSdk) {
+        const appsSdkSafeInfo = await safeAppsSdk.safe.getInfo()
+        setSafeInfo((prevSafeInfo) => {
+          // only update the isReadyOnly flag if we have safe info returned from the api kit
+          if (!prevSafeInfo) return
+          return {
+            ...prevSafeInfo,
+            isReadOnly: appsSdkSafeInfo.isReadOnly,
+          }
+        })
+      }
+    }
 
-    update()
+    const safeInfoInterval = setInterval(updateSafeInfo, SAFE_INFO_UPDATE_INTERVAL)
+    const safeAppsSdkInfoInterval = setInterval(updateSafeAppsSdkInfo, SAFE_APPS_SDK_INFO_UPDATE_INTERVAL)
```

Comment:
Why not only registering the interval if its relevant? 

so `updateSafeAppsSdkInfo` is only register if `safeAppsSdk` is defined
same for `safeInfoInterval` with `chainId && account && provider`

Reply:
this is not related to this PR.